src/run_part.F90

```diff
@@ -103,15 +113,19 @@ module pmc_run_part
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 
   !> Do a particle-resolved Monte Carlo simulation.
+#ifdef PMC_USE_CAMP
+  subroutine run_part(scenario, env_state, aero_data, aero_state, gas_data, &
+       gas_state, run_part_opt, camp_core, photolysis)
+#else
   subroutine run_part(scenario, env_state, aero_data, aero_state, gas_data, &
        gas_state, run_part_opt)
-
+#endif
     !> Environment state.
     type(scenario_t), intent(in) :: scenario
     !> Environment state.
     type(env_state_t), intent(inout) :: env_state
     !> Aerosol data.
-    type(aero_data_t), intent(in) :: aero_data
+    type(aero_data_t), intent(inout) :: aero_data
```

Comment:
I haven't yet had time to carefully trace out exactly what this would look like if we move it to `aero_state`. I tend to think that's the correct solution. I don't mind introducing a pointer.

Do you want to take a go at moving this to `aero_state`? If you push a single commit that does only this change, then we can revert it if it turns out to be too horrible.

Reply:
Yep, I'll move it to aero_state and see what happens!